### DISCUSSION TITLE
Wrapping an image around a 3D object

### DISCUSSION THREAD

**Discussion by ivanamar996** [ImageMagick/ImageMagick#6323] [Help]
### ImageMagick version

7.1.0-55

### Operating system, version and so on

Windows

### Description

I am trying to wrap an image around the object by using command:

magick MO8933-03_POS1_vectormap.png -resize "2695x2710^!" MO8933-1_tex_map.png -virtual-pixel Edge -compose Distort -define compose:args=100%%x100%% -composite MO8933-03_POS1.jpg -compose Overlay -composite pntp_usr.png

I have created vector mask that represent the are on which logo should be placed.

The mask :
![MO8933-03_POS1_vectormap](https://github.com/ImageMagick/ImageMagick/assets/39298540/0d24746e-2707-4909-8a5c-9e455f9f789c)

Artwotk:
![MO8933-1_tex_map](https://github.com/ImageMagick/ImageMagick/assets/39298540/b777a11b-42e5-4011-bd44-170254884afe)

and the product image is:
![MO8933-03_POS1](https://github.com/ImageMagick/ImageMagick/assets/39298540/c2d327bf-3334-4dfd-8f07-2a1dac578980)

But for some reason the end result is not good:
![pntp_usr](https://github.com/ImageMagick/ImageMagick/assets/39298540/75a6da65-640c-4111-84e1-5fd9944743c3)

Do you maybe know what I am doing wrong? The issue is on step that am trying to create image based on vector map and artwork, but for some reason it ignores map and all colors from the logo disappears and are converted to black.

I am using image magick version:

Version: ImageMagick 7.1.0-55 Q16-HDRI x64 https://github.com/ImageMagick/ImageMagick/commit/6edfae3829240aedef8a3a79b23c2fa1fd4c688f:20221217 https://imagemagick.org/
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193431937)

Kind regads,
Ivana

**Comment by fmw42** [comment]
Try this.  Why did you double your percents? Are you running it in a .bat script?

`magick overlay.png -resize 2695x2710^! mask.png -virtual-pixel none -compose Distort -define compose:args=100%x100%% -composite background.jpg +swap -compose over -composite pntp_usr.png`

![pntp_usr](https://github.com/ImageMagick/ImageMagick/assets/13458218/9e96bc52-974b-482f-a9a5-aa8b7fb26d4f)

**Comment by snibgo** [comment]
Sadly, the question has been double-posted. See also my reply at https://github.com/ImageMagick/ImageMagick/discussions/6322

**Reply by fmw42** [reply]
Bad form by the OP.  That explains why I could not find my answer in the other post!

**Comment by ivanamar996** [comment]
Thanks a lot, I figured out what was wrong with my code.

**Comment by snibgo** [comment]
Good stuff.

It might help other people if you said what was wrong with your code.

**Comment by ivanamar996** [comment]
I've replaced the image file with the vector map in the command. First image should be logo and the second one is vector map. :)
